### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by taborzbislaw** [opensearch-project/ml-commons#3199]
I have gone through the example: opensearch-py-ml/examples/demo_deploy_cliptextmodel.html

Model is correctly registered in opensearch cluster but the final command of the example:
ml_client.deploy_model(model_id)
ends with an error Exception: Model deployment failed

Afrer trying to deploy the model using console directly on the cluster i.e.:
POST /_plugins/_ml/models/<model_id>/_deploy

and checking the corresponding task output:
GET /_plugins/_ml/tasks/<task_id>

I see that deployment fails with the following error returned by the task:
{
  "model_id": "uDNmgZIBC9ZdJM8aMbns",
  "task_type": "DEPLOY_MODEL",
  "function_name": "TEXT_EMBEDDING",
  "state": "FAILED",
  "worker_node": [
    "xbE0btjVQUGBAWrVqCmUoQ"
  ],
  "create_time": 1728748205169,
  "last_update_time": 1728748210632,
  **"error": """{"xbE0btjVQUGBAWrVqCmUoQ":"\nUnknown builtin op: aten::scaled_dot_product_attention.\nHere are some suggestions: \n\taten::_scaled_dot_product_attention\n\nThe original call is:\n  File \"code/__torch__/transformers/models/clip/modeling_clip.py\", line 190\n    key_states = torch.transpose(torch.view(_54, [_49, -1, 8, 64]), 1, 2)\n    value_states = torch.transpose(torch.view(_55, [_48, -1, 8, 64]), 1, 2)\n    attn_output = torch.scaled_dot_product_attention(query_states, key_states, value_states, attn_mask, 0., False, scale=0.125)\n                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE\n    attn_output0 = torch.transpose(attn_output, 1, 2)\n    input = torch.reshape(attn_output0, [_47, _51, _52])\n"}""",**
  "is_async": true
}

**Comment by jovanovic-milos** [comment]
Hey, im getting exactly the same issue. Then i tried to upload ONNX model type and im getting:
`{
    "model_id": "A51B-JIBVoJr-PiFnVNy",
    "task_type": "DEPLOY_MODEL",
    "function_name": "TEXT_EMBEDDING",
    "state": "FAILED",
    "worker_node": [
        "lNpYAlQ6T0Wu5ymYW4OQ4g"
    ],
    "create_time": 1730741918213,
    "last_update_time": 1730741921643,
    "error": "{\"lNpYAlQ6T0Wu5ymYW4OQ4g\":\"Error code - ORT_FAIL - message: Deserialize tensor onnx::MatMul_3915 failed.GetFileLength for /usr/share/opensearch/data/ml_cache/models_cache/models/A51B-JIBVoJr-PiFnVNy/9/intfloat/multilingual-e5-large/onnx__MatMul_3915 failed:Invalid fd was supplied: -1\"}",
    "is_async": true
}`

**Comment by jzhang2-coveo** [comment]
I have the same problem. Is there a workaround? I am using PyTorch 2.5.1 on OpenSearch 2.17.1

**Comment by taborzbislaw** [comment]
Yes, there is workaround - compute your embeddings locally and then send them to an opensearch index as parts of indexed documents.

**Comment by jzhang2-coveo** [comment]
I fixed my issue by downgrading PyTorch to 1.13.1 and Transformers to 4.32. Then use the script shared here: https://github.com/opensearch-project/ml-commons/issues/1710

**Comment by viknesh-nm** [comment]
@jzhang2-coveo 

I'm using OpenSearch version 2.19 and encountering an issue while deploying a Hugging Face model. I'm trying to register and deploy the following model:

```
curl -XPUT "https://opensearch-node1/_cluster/settings" -H 'Content-Type: application/json' -d'
{
  "persistent": {
    "plugins.ml_commons.only_run_on_ml_node": "false",
    "plugins.ml_commons.model_access_control_enabled": "true",
    "plugins.ml_commons.native_memory_threshold": "99"
  }
}'
```


```
curl -XPOST "https://opensearch-node1:9200/_plugins/_ml/models/_register?deploy=true" -H 'Content-Type: application/json' -d'
{
  "name": "huggingface/sentence-transformers/all-MiniLM-L6-v2",
  "version": "1.0.2",
  "model_format": "TORCH_SCRIPT"
}'
```


```curl -XGET "https://opensearch-node1:9200/_plugins/_ml/tasks/{task_id_from_the_above_response}"```


```curl -XGET "https://opensearch-node1:9200/_plugins/_ml/models/{model_id_from_the_above_response}"```

FInal Output:
```
{
  "name": "huggingface/sentence-transformers/all-MiniLM-L6-v2",
  "model_group_id": "5-cOnpcBjCUwxHPZCaeR",
  "algorithm": "TEXT_EMBEDDING",
  "model_version": "2",
  "model_format": "TORCH_SCRIPT",
  "model_state": "DEPLOY_FAILED",
  "model_content_size_in_bytes": 91789778,
  "model_content_hash_value": "25e2858993cd477936f24e412a508b005aa6b59a308301cc69690e4b90cab439",
  "model_config": {
    "model_type": "bert",
    "embedding_dimension": 384,
    "framework_type": "SENTENCE_TRANSFORMERS",
    "all_config": """{"_name_or_path": "sentence-transformers/all-MiniLM-L6-v2", "architectures": ["BertModel"], "attention_probs_dropout_prob": 0.1, "classifier_dropout": null, "gradient_checkpointing": false, "hidden_act": "gelu", "hidden_dropout_prob": 0.1, "hidden_size": 384, "initializer_range": 0.02, "intermediate_size": 1536, "layer_norm_eps": 1e-12, "max_position_embeddings": 512, "model_type": "bert", "num_attention_heads": 12, "num_hidden_layers": 6, "pad_token_id": 0, "position_embedding_type": "absolute", "torch_dtype": "float32", "transformers_version": "4.49.0", "type_vocab_size": 2, "use_cache": true, "vocab_size": 30522}""",
    "pooling_mode": "MEAN",
    "normalize_result": true
  },
  "created_time": 1750703696719,
  "last_updated_time": 1750703703762,
  "last_registered_time": 1750703702597,
  "last_deployed_time": 1750703703761,
  "total_chunks": 10,
  "planning_worker_node_count": 2,
  "current_worker_node_count": 0,
  "planning_worker_nodes": [
    "0fOfctooRq6MC4fqLie3gA",
    "8jjGyGQyQESje2hIgA20AQ"
  ],
  "deploy_to_all_nodes": true,
  "is_hidden": false
}
```


I see the model output as `DEPLOY_FAILED`, and the log says 

```
attn_output = torch.scaled_dot_product_attention(query_layer, key_layer, value_layer, attention_mask)
```


More info for logs

```
[2025-06-23T18:35:03,739][ERROR][o.o.m.e.a.DLModel        ] [opensearch-node1] Failed to deploy model s3YSnpcBN3WlkcyFazNR

ai.djl.engine.EngineException: 

Unknown builtin op: aten::scaled_dot_product_attention.

Here are some suggestions: 

	aten::_scaled_dot_product_attention


The original call is:

  File "code/__torch__/transformers/models/bert/modeling_bert.py", line 165

    x1 = torch.view(_36, [_38, int(_39), 12, 32])

    value_layer = torch.permute(x1, [0, 2, 1, 3])

    attn_output = torch.scaled_dot_product_attention(query_layer, key_layer, value_layer, attention_mask)

                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE

    attn_output0 = torch.transpose(attn_output, 1, 2)

    input = torch.reshape(attn_output0, [_26, _27, 384])
```

**Additional Notes:**
- I'm testing via docker-compose
- This works fine in OpenSearch 3.0.
- I also tried it in version 2.17, but faced the same issue.
- Upgrading to 3.0 is not an option right now for me, as it's not yet available in AWS OpenSearch.



Could you please confirm if there's a workaround or if this op will be supported in a future 2.x patch? Any suggestions for alternative deployment options on 2.19 would be appreciated.
